### PR TITLE
Create slack notification step for master builds

### PR DESCRIPTION
**Pull request by jamesmunro**
From here: https://github.com/integrations/slack/issues/1563#issuecomment-1588009077

Uses this integration: https://github.com/ravsamhq/notify-slack-action

### CODE REVIEW

**Review by maxim-morozov — APPROVED** [review]

File: .github/workflows/failure_notification.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Check for master failure
+on:
+  workflow_run:
+    workflows: [Build and Test]
```
Comment (poodlewars) [inline .github/workflows/failure_notification.yaml:4]: Why not the Conda or Analysis builds?

**Comment by poodlewars** [comment]
Needs test evidence showing a notification for an example branch

File: .github/workflows/failure_notification.yaml (hunk 1 above)
Comment (jamesmunro) [reply .github/workflows/failure_notification.yaml:4]: Yes, can do.  Just wanted to start with something.

**Review by poodlewars — APPROVED** [review]

**Comment by jamesmunro** [comment]
> Needs test evidence showing a notification for an example branch
It's not possible to test this type of trigger outside the default branch.
https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#workflow_run
image

**Pull request merged**